packages/circuits/src/aadhaar-verifier.circom

```diff
@@ -1,111 +1,73 @@
 pragma circom 2.1.6;
 
-include "./helpers/rsa.circom";
-include "./helpers/sha.circom";
-include "./helpers/timestamp.circom";
-include "./extractor.circom";
-
-// Circuit to verify Aadhaar signature
-// n: RSA pubic key size per chunk
-// k: Number of chunks the RSA public key is split into
-// maxDataLength: Maximum length of the data
+include "circomlib/circuits/poseidon.circom";
+include "./helpers/signature.circom";
+include "./helpers/nullifier.circom";
+include "./helpers/extractor.circom";
+
+
+/// @title AadhaarVerifier
+/// @notice This circuit verifies the Aadhaar QR data using RSA signature
+/// @param n - RSA pubic key size per chunk
+/// @param k - Number of chunks the RSA public key is split into
+/// @param maxDataLength - Maximum length of the data
+/// @input qrDataPadded - QR data without the signature; each number represent ascii byte; remaining space is padded with 0
+/// @input qrDataPaddedLength - Length of padded QR data
+/// @input nonPaddedDataLength - Length of actual data without padding
+/// @input delimiterIndices - Indices of delimiters (255) in the QR text data. 18 delimiters including photo
+/// @input signature - RSA signature
+/// @input pubKey - RSA public key (of the government)
+/// @input appId - Application ID which will be included in identityNullifier
+/// @input public signalHash - An external signal to make it part of the proof
+/// @output identityNullifier - PosidonHash(name, dob, gender)
+/// @output userNullifier - PosidonHash(photo)
+/// @output timestamp - Timestamp of when the data was signed - extracted and converted to Unix timestamp
+/// @output pubkeyHash - Poseidon hash of the RSA public key
 template AadhaarVerifier(n, k, maxDataLength) {
-    signal input aadhaarData[maxDataLength];    // Aadhaar data padded (the data that is SHA hashed and signed)
-    signal input aadhaarDataLength;             // length of the padded data
-    signal input signature[k];                  // RSA signature
-    signal input pubKey[k];                     // RSA public key (of the government)
+    signal input qrDataPadded[maxDataLength];
+    signal input qrDataPaddedLength;
+    signal input nonPaddedDataLength;
+    signal input delimiterIndices[18];
+    signal input signature[k];
+    signal input pubKey[k];
+    signal input appId;
```

Comment:
Don't you want to change to `nullifierSeed`?

Reply:
Lets do that in another PR. Its not just about the rename I think - we will need to make this input private and then add another output which is the hash(nullifierSeed).
So, lets review the idea again and do in another PR.

Reply:
All right!